kernel/src/actions/mod.rs

```diff
@@ -297,9 +304,12 @@ impl Protocol {
     /// support the specified protocol writer version and all enabled writer features?
     pub(crate) fn ensure_write_supported(&self) -> DeltaResult<()> {
         match &self.writer_features {
-            Some(writer_features) if self.min_writer_version == 7 => {
-                // if we're on version 7, make sure we support all the specified features
-                ensure_supported_features(writer_features, &SUPPORTED_WRITER_FEATURES)
+            // if min_reader_version = 3 and min_writer_version = 7 and all writer features are
+            // supported => OK
+            Some(writer_features)
+                if self.min_reader_version == 3 && self.min_writer_version == 7 =>
+            {
+                ensure_supported_feature(writer_features, &SUPPORTED_WRITER_FEATURES)
```

Comment:
hm. I think we originally attempted the insulate the read/write support here - though I guess the protocol states that 3 AND 7 are required for table features to be valid?

regardless I think this should be in a separate PR just so we aren't combining too many changes here.